rfd/0091-non-ad-desktop-access.md

```diff
@@ -0,0 +1,144 @@
+---
+authors: Przemko Robakowski (przemko.robakowski@goteleport.com)
+state: draft
+---
+
+# RFD 91 - Desktop access to hosts that are not connected to Active Directory
+
+## Required Approvers
+
+- Engineering: @zmb3 && (@ibeckermayer || @LKozlowski)
+- Product: (@klizhentas || @xinding33)
+
+## Overview
+
+Common ask for Desktop Access in Teleport is to connect to hosts that are not part of the domain, so there's no Active
+Directory available and no Kerberos involved. Unfortunately Windows supports smart card login, used currently in
+Teleport, only for domain-joined hosts, so we need another solution for non-AD case.
+
+Idea is to use install on host DLL library containing custom
+[authentication package](https://learn.microsoft.com/en-us/windows/win32/secauthn/windows-authentication-packages). This
+authentication package will (in combination with custom credential provider) check certificate on smart card and will
+log user in.
+
+## Details
+
+```
+
+                 RDP
+         ---------------------                         
+         |                   |                         
++------------------+     +------------------+                +------------------+ 
+|                  |     | Windows Desktop  |--------------->|    Credential    | 
+|    Teleport      |     |                  |                |     provider     | 
+-  Windows Desktop ------|   Winlogon.exe   |   Credential   |                  | 
+|     Service      |     |                  |<---------------|  Verify cert     | 
++------------------+     +------------------+                +------------------+ 
+                            ^  |            +-------------------+     +---------+
+                            |  | Credential |   Authentication  |     |         |           
+                            |  |----------->|      package      |<--->|   SAM   |
+                            |               |                   |     |         |           
+                            |     Token     | Verify credential |     +---------+           
+                            |---------------|Create login token |                             
+                                            +-------------------+                             
```

Comment:
This diagram is slightly confusing, if I'm understanding it correctly we're looking at something like this (?):

<img width="704" alt="Screen Shot 2022-10-06 at 17 12 28" src="https://user-images.githubusercontent.com/13578537/194441103-beb8aa6c-11c7-4980-bbe9-8983d4b86222.png">